ospfd/ospf_sr.c

```diff
@@ -639,26 +609,114 @@ static int ospf_zebra_send_mpls_labels(int cmd, struct sr_nhlfe nhlfe)
 	return zebra_send_mpls_labels(zclient, cmd, &zl);
 }
 
-/* Add new NHLFE entry for SID */
-static inline void add_sid_nhlfe(struct sr_nhlfe nhlfe)
+/* Add new NHLFE entry for Adjacency SID */
+static inline void add_adj_sid(struct sr_nhlfe nhlfe)
 {
-	if ((nhlfe.label_in != 0) && (nhlfe.label_out != 0))
-		ospf_zebra_send_mpls_labels(ZEBRA_MPLS_LABELS_ADD, nhlfe);
+	if (nhlfe.label_in != 0)
+		send_adjacency_sid(ZEBRA_MPLS_LABELS_ADD, nhlfe);
 }
 
-/* Remove NHLFE entry for SID */
-static inline void del_sid_nhlfe(struct sr_nhlfe nhlfe)
+/* Remove NHLFE entry for Adjacency SID */
+static inline void del_adj_sid(struct sr_nhlfe nhlfe)
 {
-	if ((nhlfe.label_in != 0) && (nhlfe.label_out != 0))
-		ospf_zebra_send_mpls_labels(ZEBRA_MPLS_LABELS_DELETE, nhlfe);
+	if (nhlfe.label_in != 0)
+		send_adjacency_sid(ZEBRA_MPLS_LABELS_DELETE, nhlfe);
+}
+
+/* Update NHLFE entry for Adjacency SID */
+static inline void update_adj_sid(struct sr_nhlfe n1, struct sr_nhlfe n2)
+{
+	del_adj_sid(n1);
+	add_adj_sid(n2);
+}
+
+/* Update NHLFE for Prefix SID */
+static void update_prefix_sid(const struct sr_prefix *srp)
+{
+	struct zapi_labels zl;
+	struct zapi_nexthop *znh;
```

Comment:
I'd prefer that the zapi messaging be in ospf_zebra.c - is there a particular reason to pull that into the sr module?

Reply:
This is historic. All functions that configure label were written inside the ospfd/ospf_sr.c file. But, I could take the opportunity of this PR to move them in ospfd/ospf_zebra.c file if this make sens